include/DcmTrajectoryGenerator.h

```diff
@@ -0,0 +1,391 @@
+/**
+ * @file DcmTrajectoryGenerator.h
+ * @author Giulio Romualdi
+ * @copyright 2018 iCub Facility - Istituto Italiano di Tecnologia
+ *            Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ * @date 2018
+ */
+
+#ifndef DCMTRAJECTORYGENERATOR_H
+#define DCMTRAJECTORYGENERATOR_H
+
+// std
+#include <vector>
+#include <memory>
+#include <deque>
+
+// Eigen3
+#include <Eigen/Dense>
+
+// iDynTree
+#include "iDynTree/Core/VectorFixSize.h"
+
+#include "FootPrint.h"
+
+# define MIN_DURATION_DOUBLE_SUPPORT_STANCE_PHASE 0.001
+
+/**
+ * Virtual class that represents a coordiante (x or y) of the Divergent Component
```

Comment:
The first line of the doxygen comment of a class or a function is typically transformed (if the [`JAVADOC_AUTOBRIEF`](https://www.stack.nl/~dimitri/doxygen/manual/config.html#cfg_javadoc_autobrief) option) option is set to `ON`) in the brief description of the class or the method, so it is tipically good to have them self-contained.